kong/plugins/rate-limiting/handler.lua

```diff
@@ -104,6 +111,18 @@ function RateLimitingHandler:access(conf)
         ngx.header[RATELIMIT_LIMIT .. "-" .. k] = v.limit
         ngx.header[RATELIMIT_REMAINING .. "-" .. k] = math.max(0, (stop == nil or stop == k) and v.remaining - 1 or v.remaining) -- -increment_value for this current request
       end
+
+      -- Find the closer limit.
+      local closer_limit = 0
+      for k, v in pairs(usage) do
```

Comment:
This loops is unnecessary, we could simply add its body to the above one.

Reply:
I created a different loop just to isolate better the patch. It's fine to integrate that in the loop ;)

Reply:
Ok, I just didn't want to modify the actual code. It's fine to modify the loop.

Reply:
done in ff678caf5